File: SmokeSaver/Models/RecoveryMilestone.cs
namespace SmokeSaver.Models;

public enum RecoveryMilestoneState
{
    Completed,
    Current,
    Upcoming
}

public sealed class RecoveryMilestone
{
    public TimeSpan Threshold { get; init; }

    public string TimeframeLabel { get; init; } = string.Empty;

    public string Title { get; init; } = string.Empty;

    public string Description { get; init; } = string.Empty;
}

public sealed class RecoveryMilestoneSnapshot
{
    public required RecoveryMilestone Milestone { get; init; }

    public RecoveryMilestoneState State { get; init; }
}

File: NitKotin.Android/app/src/main/java/com/nitkotin/android/data/model/RecoveryMilestone.kt
package com.nitkotin.android.data.model

import java.time.Duration

enum class RecoveryMilestoneState {
    Completed,
    Current,
    Upcoming,
}

data class RecoveryMilestone(
    val threshold: Duration,
    val timeframeLabel: String,
    val title: String,
    val description: String,
)

data class RecoveryMilestoneSnapshot(
    val milestone: RecoveryMilestone,
    val state: RecoveryMilestoneState,
)
